src/pool/pool_disjoint_internal.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_POOL_DISJOINT_INTERNAL_H
+#define UMF_POOL_DISJOINT_INTERNAL_H 1
+
+#include <assert.h>
+#include <ctype.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <umf/memory_pool.h>
+#include <umf/memory_pool_ops.h>
+#include <umf/memory_provider.h>
+#include <umf/pools/pool_disjoint.h>
+
+#include "critnib/critnib.h"
+#include "uthash/utlist.h"
+
+#include "base_alloc_global.h"
+#include "provider/provider_tracking.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+#include "utils_log.h"
+#include "utils_math.h"
+#include "utils_sanitizers.h"
+
+typedef struct bucket_t bucket_t;
+typedef struct slab_t slab_t;
+typedef struct slab_list_item_t slab_list_item_t;
+typedef struct disjoint_pool_t disjoint_pool_t;
+
+typedef struct bucket_t {
+    size_t size;
+
+    // Linked list of slabs which have at least 1 available chunk.
+    // We always count available slabs as an optimization.
+    slab_list_item_t *available_slabs;
+    size_t available_slabs_num;
+
+    // Linked list of slabs with 0 available chunk.
+    slab_list_item_t *unavailable_slabs;
+
+    // Protects the bucket and all the corresponding slabs
+    utils_mutex_t bucket_lock;
+
+    // Reference to the allocator context, used access memory allocation
+    // routines, slab map and etc.
+    disjoint_pool_t *pool;
+
+    umf_disjoint_pool_shared_limits_handle_t shared_limits;
+
+    // For buckets used in chunked mode, a counter of slabs in the pool.
```

Comment:
Mixing description of chunked slabs with non-chunked brings confusion in this comment. Please separate these cases, i.e. by creating two blocks of text, one for each case.